product_weight/wizard/product_weight_update.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2013 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+#    Copyright (C) 2015 Akretion (<http://www.akretion.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductWeightUpdate(models.TransientModel):
+    _name = "product.weight.update"
+    _description = "Update Product Weight"
+
+    product_tmpl_id = fields.Many2one('product.template', 'Product')
+    bom_id = fields.Many2one(
+        'mrp.bom',
+        'BoM',
+        domain="[('product_tmpl_id', '=', product_tmpl_id)]")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(ProductWeightUpdate, self).default_get(fields)
+        if not fields:
+            return res
+        if self._context.get('active_model') == 'product.template':
+            product_tmpl_id = self._context and \
+                self._context.get('active_id', False) or False
+            domain = [('product_tmpl_id', '=', product_tmpl_id)]
+        else:
+            product_id = self._context and \
+                self._context.get('active_id', False) or False
```

Comment:
_context is a private variable here, in my opinion, you should avoid using it.

Also, the following expression is redondant:
product_id = self._context and self._context.get('active_id', False) or False
You may change it for
context = self.env.context
...
product_id = context.get('active_id', False)